yellow-paper/docs/addresses-and-keys/addresses-and-keys.md

```diff
@@ -0,0 +1,903 @@
+
+
+\gdef\d{\color{green}{d}}
+\gdef\Gd{\color{green}{G_d}}
+
+\gdef\Ivpkappd{\color{violet}{Ivpk_{app,d}}}
+\gdef\Rivpkappd{\color{violet}{Rivpk_{app,d}}}
+\gdef\Pkappd{\color{violet}{Pk_{app,d}}}
+\gdef\shareableIvpkappd{\color{violet}{\widetilde{Ivpk_{app,d}}}}
+
+
+\gdef\ivskappstealth{\color{red}{ivsk_{app,stealth}}}
+\gdef\Ivpkappdstealth{\color{violet}{Ivpk_{app,d,stealth}}}
+\gdef\Pkappdstealth{\color{violet}{Pk_{app,d,stealth}}}
+\gdef\shareableivskappstealth{\color{orange}{\widetilde{ivsk_{app,stealth}}}}
+\gdef\shareableIvpkappdstealth{\color{violet}{\widetilde{Ivpk_{app,d,stealth}}}}
+
+\gdef\hstealth{\color{violet}{h_{stealth}}}
+
+
+\gdef\esk{\color{red}{esk}}
+\gdef\Epk{\color{green}{Epk}}
+\gdef\Epkd{\color{green}{Epk_d}}
+\gdef\eskheader{\color{red}{esk_{header}}}
+\gdef\Epkheader{\color{green}{Epk_{header}}}
+\gdef\Epkdheader{\color{green}{Epk_{d,header}}}
+
+\gdef\sharedsecret{\color{violet}{\text{S}}}
+\gdef\sharedsecretmheader{\color{violet}{\text{S_{m,header}}}}
+\gdef\sharedsecretappheader{\color{violet}{\text{S_{app,header}}}}
+
+
+\gdef\hmencheader{\color{violet}{h_{m,enc,header}}}
+\gdef\happencheader{\color{violet}{h_{app,enc,header}}}
+\gdef\hmenc{\color{violet}{h_{m,enc}}}
+\gdef\happenc{\color{violet}{h_{app,enc}}}
+\gdef\incomingenckey{\color{violet}{h_{incoming\_enc\_key}}}
+
+
+\gdef\plaintext{\color{red}{\text{plaintext}}}
+\gdef\ciphertext{\color{green}{\text{ciphertext}}}
+\gdef\ciphertextheader{\color{green}{\text{ciphertext\_header}}}
+\gdef\payload{\color{green}{\text{payload}}}
+
+
+\gdef\tagg{\color{green}{\text{tag}}}
+\gdef\Taghs{\color{green}{\text{Tag}_{hs}}}
+
+$$
+
+# Proposal: Aztec Keys
+
+
+## Requirements for Keys
+
+:::info Disclaimer
+This is a draft. These requirements need to be considered by the wider team, and might change significantly before a mainnet release.
+
+Eventually, the key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document should align with the interpretations as described in RFC 2119
+:::
+
+These requirements take priority over optimisations.
+
+Note: there is nothing to stop an app and wallet from implementing its own key derivation scheme. Nevertheless, we're designing a 'canonical' scheme that most developers and wallets can use.
+
+### Authorisation keys
+
+Tx authentication is done via a custom account contract, so authorisation keys aren't specified at the protocol level.
+
+A tx authentication secret key is arguably the most important key to keep private, because knowledge of such a key could potentially enable an attacker to impersonate the user and execute a variety of functions on the network. 
+
+Requirements:
+- A tx authentication secret key SHOULD NOT enter Aztec software, and SHOULD NOT enter a circuit.
+  - Reason: this is just best practice.
+
+
+### Master & Siloed Keys
+
+Requirements:
+- All keys must be re-derivable from a single `seed` secret.
```

Comment:
Yeah, a secret can be derived in loads of ways. And you're right, Aztec Connect had the user sign a string `""Sign this message to generate your Aztec Privacy Key. This key lets the  application decrypt your balance on Aztec.\n\nIMPORTANT: Only sign this  message if you trust the application."`, signed using the user's Ethereum private key.
The problem with this approach is a malicious app could trick a user into signing this string again, and then steal funds.

Reply:
> The problem with this approach is a malicious app could trick a user into signing this string again, and then steal funds

I see, but that's a problem that can be avoided in an Aztec-specific wallet. So it's "cryptographically" safe to recommend that this seed is derived from the signing key (assuming there's one), so Aztec users only need to store a single piece of key material?

Reply:
Potentially - it might depend on the signature scheme. Some signature schemes make use of randomness, so the signature looks different each time (so would be a bad way to derive a seed, unless you kept a record of the randomness, or something...)